src/mover_policies_containers_creation_callback.erl

```diff
@@ -0,0 +1,432 @@
+	 error_halts_migration/0,
+	 reconfigure_object/2,
+	 migration_action/2,
+	 next_object/0
+	]).
+
+-include("mover.hrl").
+-include("mv_oc_chef_authz.hrl").
+
+-record(mover_org, {id, authz_id, name, superuser}).
+-record(mover_requestor, {id, authz_id, username}).
+
+%% This is an ACE in human readable form
+-record(mover_hr_ace, {clients = [],
+                 users = [],
+                 groups = []}).
+
+-record(mover_chef_container, {
+          id,
+          authz_id,
+          org_id,
+          name,
+          last_updated_by,
+          created_at,
+          updated_at
+         }).
+
+-define(ORG_POLICY_FOR_POLICIES_CONTAINERS,
+        [{create_containers, [policies, policy_groups, cookbook_artifacts]},
+
+         {acls,
+          [
+           %% Creator (superuser normally) goes everywhere
+           {add_acl, [{container, policies}],           [create, read, update, delete, grant], [{user, creator}]},
+           {add_acl, [{container, policy_groups}],      [create, read, update, delete, grant], [{user, creator}]},
+           {add_acl, [{container, cookbook_artifacts}], [create, read, update, delete, grant], [{user, creator}]},
+
+           %% Admins
+           {add_acl, [{container, policies}],           [create, read, update, delete, grant], [{group, users}]},
+           {add_acl, [{container, policy_groups}],      [create, read, update, delete, grant], [{group, users}]},
+           {add_acl, [{container, cookbook_artifacts}], [create, read, update, delete, grant], [{group, users}]},
+
+
+           %% users
+           {add_acl, [{container, policies}],           [create, read, update, delete], [{group, users}]},
+           {add_acl, [{container, policy_groups}],      [create, read, update, delete], [{group, users}]},
+           {add_acl, [{container, cookbook_artifacts}], [create, read, update, delete], [{group, users}]},
+
+           %% clients
+
+           {add_acl, [{container, policies}],           [read], [{group, clients}]},
+           {add_acl, [{container, policy_groups}],      [read], [{group, clients}]},
+           {add_acl, [{container, cookbook_artifacts}], [read], [{group, clients}]}
+          ]
+         }
+        ]).
+
+migration_init() ->
+    %% TODO: remove this when testing is complete:
+    Nuke = <<"delete from containers where name in ('policies', 'policy_groups', 'cookbook_artifacts');">>,
+    sqerl:execute(Nuke),
+
+    %% TODO: this fails if the pool is already created. Should be tolerant of
+    %% errors and also should tear this down after the migration
+    mv_oc_chef_authz_http:create_pool(),
+    %% TODO: superuser name is somewhat configurable, it's set in oc_erchef's sys.config
+    {ok, SuperuserResults} = sqerl:adhoc_select([username, id, authz_id], users, {username, equals, <<"pivotal">>}),
+    Superuser = db_results_to_requestor(hd(SuperuserResults)),
+
+    %% Should do this for each container type and uniq the orgs just in case a migration partially bombs (?)
```

Comment:
If a migration fails to create one of the 3 containers, it can never succeed in the future, since we'll blow up later when we try again to create it. Should I:

* not worry about it
* split this into 3 container creation upgrades
* make the container creation try to handle that case (may be very tricky if the authz permissions don't get properly created
* pre-emptively nuke the containers (e.g., `delete from containers where name in (policies, policy_groups, cookbook_artifacts) and org_id = $org_id`) at the beginning of each org create.

Thoughts?